src/test/scala/mesosphere/marathon/core/health/impl/AppHealthCheckActorTest.scala

```diff
@@ -0,0 +1,133 @@
+package mesosphere.marathon
+package core.health.impl
+
+import akka.testkit.TestProbe
+import mesosphere.AkkaUnitTest
+import mesosphere.marathon.core.event.InstanceHealthChanged
+import mesosphere.marathon.core.health.impl.AppHealthCheckActor.{AddHealthCheck, ApplicationKey, HealthCheckStatusChanged, RemoveHealthCheck}
+import mesosphere.marathon.core.health.{Health, MarathonHttpHealthCheck, PortReference}
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.state.PathId._
+import mesosphere.marathon.state.Timestamp
+
+class AppHealthCheckActorTest extends AkkaUnitTest {
+  class Fixture {
+    val appId = "/test".toPath
+    val appVersion = Timestamp(1)
+    val appKey = ApplicationKey(appId, appVersion)
+    val hcPort80 = MarathonHttpHealthCheck(portIndex = Some(PortReference(80)))
+    val hcPort443 = MarathonHttpHealthCheck(portIndex = Some(PortReference(443)))
+    val instances = List(
+      Instance.Id("instance1"),
+      Instance.Id("instance2"),
+      Instance.Id("instance3")
+    )
+  }
+
+  "AppHealthCheckActor" should {
+    "send status changed event when all instances are healthy" in {
+      val f = new Fixture
+      val systemLog = TestProbe()
+
+      val actor = system.actorOf(AppHealthCheckActor.props(system.eventStream))
+      system.eventStream.subscribe(systemLog.ref, classOf[InstanceHealthChanged])
+
+      actor ! AddHealthCheck(f.appKey, f.hcPort80)
+      actor ! AddHealthCheck(f.appKey, f.hcPort443)
+
+      actor ! HealthCheckStatusChanged(f.appKey, f.hcPort80,
+        Health(f.instances.head, lastSuccess = Some(Timestamp(5)), lastFailure = Some(Timestamp(0))))
+      actor ! HealthCheckStatusChanged(f.appKey, f.hcPort443,
+        Health(f.instances.head, lastSuccess = Some(Timestamp(5)), lastFailure = Some(Timestamp(0))))
+
+      actor ! RemoveHealthCheck(f.appKey, f.hcPort80)
+      actor ! RemoveHealthCheck(f.appKey, f.hcPort443)
```

Comment:
Can we verify that the actor has cleaned up the entries for this appKey? Use akka testkit underlyingActor method to access the internal map directly?

Reply:
I'll check how I can do it with what you propose.